components/scream/src/control/surface_coupling.cpp

```diff
@@ -148,13 +148,13 @@ register_export (const std::string& fname,
   if (m_field_mgr->has_field(fname)) {
 
     // If fname is a field in the the field manager, then set the data and column info based on this field.
-    // Note that the field must not have a parent as get_view() would be unsafe.
+    // Note that the field must not have a parent as get_flattened_view() would be unsafe.
```

Comment:
This comment is actually incorrect since we allow for fields with a parent. Got left in I guess from a previous version of SurfaceCoupling that I cycled through. Could you remove it? Just the part `// Note that the field...`

Reply:
Ah, right. I will have to rework the register method then, since I don't want to allow `get_flattened_view()` for subfields. Will it work if I extract the parent, and  grab the pointer from that? Meaning, does the current impl of `get_col_info` correctly offset a subfield from the start _of the parent field's_ view?

Reply:
Yes, `get_col_info` assumes that a `subview.data()` is exactly the `parent_view.data()`. So something like
```
if (field has a parent) {
  info.data = parent_field.get_flattened_view.data();
} else {
  info.data = field.get_flattened_view.data();
}
```
should work.